torch/_dynamo/output_graph.py

```diff
@@ -464,8 +466,9 @@ def guards(self) -> Set[Guard]:
     def nn_modules(self) -> Dict[str, torch.nn.Module]:
         return self.tracing_context.module_context.nn_modules
 
-    def save_global_state(self):
-        global_state = self.tracing_context.global_context.global_state
+    def save_global_state(self, global_state=None):
```

Comment:
This argument is counterintuitive: you pass it a fresh dictionary to be mutated

Reply:
How about `self.save_global_state(out=current_global_state)`

Reply:
changed to `out=None`